main_test.go

```diff
@@ -544,6 +605,48 @@ func TestGetVMName(t *testing.T) {
 	}
 }
 
+func TestCheckIPv6Capable(t *testing.T) {
+	tests := []struct {
+		desc        string
+		httpHandler func(http.ResponseWriter, *http.Request)
+		wantOutput  bool
+	}{
+		{
+			desc: "v6 enabled",
+			httpHandler: func(w http.ResponseWriter, r *http.Request) {
+				if r.Header.Get("Metadata-Flavor") != "Google" {
+					http.Error(w, "Missing Metadata-Flavor", 403)
+					return
+				}
+				w.Write([]byte("6970:7636:2061:6464:7265:7373:2062:6162"))
+			},
+			wantOutput: true,
+		},
+		{
+			desc: "v6 not enabled",
+			httpHandler: func(w http.ResponseWriter, r *http.Request) {
+				http.Error(w, "Not Found", 404)
+				return
+			},
+			wantOutput: false,
+		},
+	}
+	server := httptest.NewServer(nil)
+	defer server.Close()
+	overrideHTTP(server)
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			http.DefaultServeMux = new(http.ServeMux)
```

Comment:
Do we still need this? It doesn't feel right to me to overwrite the package level variable `DefaultServeMux`.

If we really want to do something like this, it might involve a few more steps I think. The `httptest` package does not support using a serveMux other than the default one. So, we might have to create our own `net.Listener`, and pass it to [Serve](https://pkg.go.dev/net/http#Serve), along with a `Handler`. Would be nice if we can avoid all of that.

Reply:
I had to do that because it throws an error since im trying to register a different handlerfunc for the same endpoint..

but this works

```
mux := new(http.ServeMux)
mux.HandleFunc("metadata.google.internal/computeMetadata/v1/instance/network-interfaces/0/ipv6s", test.httpHandler)
server := httptest.NewServer(mux)
defer server.Close()
overrideHTTP(server)
if got := isIPv6Capable(); got != test.wantOutput {
 t.Fatalf("isIPv6Capable() = %t, want: %t", got, test.wantOutput)
}
```

lemeno if this is okayy